project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:convert';
+import 'package:collaborative_science_platform/utils/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+class SettingsProvider with ChangeNotifier {
+  String token = "cd018223f0564aa0f66af496d754c819494de87f"; //TEST
+  Future<void> changePassword(String oldPass, String newPass) async {
```

Comment:
Here changePassword should take User user as argument.
changePassword(User user, String oldPass, String newPass)